src/backend/core/api/viewsets.py

```diff
@@ -373,6 +380,8 @@ def get_queryset(self):
                 # Abilities are computed based on logged-in user's role and
                 # the user role on each team access
                 .annotate(user_role=Subquery(user_role_query))
+                .annotate(email=Subquery(user_main_identity.values("email")[:1]))
+                .annotate(name=Subquery(user_main_identity.values("name")[:1]))
```

Comment:
I could have done also smth as :
```
    .annotate(
        email=Max(
            "user__identities__email",
            filter=Q(user__identities__is_main=True),
        )
    )
    .annotate(
        name=Max(
            "user__identities__name",
            filter=Q(user__identities__is_main=True),
        )
    )

```
lmk what do you prefer

Reply:
Also, please let consider a future refactoring. We are joining twice the user-related data, using both annotate and prefetch. Could we do a simple join?

Reply:
This is what is was thinking about the other day : [order_by at the same moment than the prefetch_related](https://stackoverflow.com/questions/48071397/django-prefetch-relateds-prefetch-order-by). I'm not sure if we can use it to avoid another annotate or query but I wanted to share it to you.

Reply:
> I could have done also smth as :
> 
> ```
>     .annotate(
>         email=Max(
>             "user__identities__email",
>             filter=Q(user__identities__is_main=True),
>         )
>     )
>     .annotate(
>         name=Max(
>             "user__identities__name",
>             filter=Q(user__identities__is_main=True),
>         )
>     )
> ```
> 
> lmk what do you prefer

That's nice too but the "Max" kinda confuses me. Does that remove 2 queries from subqueries, though ? or are they already "included" in the "main" query ?

Reply:
> This is what is was thinking about the other day : [order_by while prefetching](https://stackoverflow.com/questions/48071397/django-prefetch-relateds-prefetch-order-by). I'm not sure if we can use it to avoid another annotate or query but I wanted to share it to you.

Let's discuss it orally, I feel the order_by is applied on the second query made to prefect data.

Reply:
Yeah, the Max is quite confusing. Actually the Subqueries don't add any query! I tried this approach, because I felt the SQL query under the hood was easier to understand. Let's stick to my initial approach

Reply:
+1 on @mjeammet's opinion that Max is confusing.